runtime/service/src/main/java/org/apache/polaris/service/context/RealmContextFilter.java

```diff
@@ -46,19 +50,20 @@ public Uni<Response> resolveRealmContext(ContainerRequestContext rc) {
                     rc.getHeaders()::getFirst))
         .onItem()
         .invoke(realmContext -> rc.setProperty(REALM_CONTEXT_KEY, realmContext))
+        // ContextLocals is used by RealmIdTagContributor to add the realm id to metrics
         .invoke(realmContext -> ContextLocals.put(REALM_CONTEXT_KEY, realmContext))
```

Comment:
Trying to understand why `RealmIdTagContributor` cannot read from the `ContainerRequestContext` instead of relying on `ContextLocals` to pass around any object needed for a `HttpServerMetricsTagsContributor`. Or would it be reasonable to copy the whole `rc` into a `ContextLocals` by default? Would you mind sharing your thought on it?